src/drawers/drawer.ts

```diff
@@ -36,16 +36,20 @@ export module Drawers {
      * Constructs a Drawer
      *
      * @constructor
-     * @param{string} key The key associated with this Drawer
+     * @param {Dataset} dataset The dataset associated with this Drawer
      */
-    constructor(key: string) {
-        this.key = key;
+    constructor(dataset: Dataset) {
+        this._dataset = dataset;
     }
 
     public setup(area: D3.Selection) {
       this._renderArea = area;
     }
 
+    protected dataset() {
```

Comment:
Why not just have the field be `protected`? Also the method should be underscore-named right?